internal/connector/internal/api/private/api_connector_clusters_agent.go

```diff
@@ -499,6 +499,240 @@ func (a *ConnectorClustersAgentApiService) GetClusterAsignedConnectorNamespaces(
 	return localVarReturnValue, localVarHTTPResponse, nil
 }
 
+/*
+GetClusterAssignedProcessorDeploymentById Returns a processor deployment assigned to the cluster.
+Returns a processor deployment assigned to the cluster.
+  - @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
+  - @param connectorClusterId The id of the connector cluster
+  - @param processorDeploymentId The id of the processor deployment
+
+@return ProcessorDeployment
+*/
+func (a *ConnectorClustersAgentApiService) GetClusterAssignedProcessorDeploymentById(ctx _context.Context, connectorClusterId string, processorDeploymentId string) (ProcessorDeployment, *_nethttp.Response, error) {
```

Comment:
@manstis What's the use-case for retrieving a specific `ProcessorDeployment` by `connectorClusterId` and `processorDeploymentId`?

Reply:
`Processors` are deployed to the same `namespace`/`cluster` as the `Connector`'s to which they can integrate.

There is an API to retrieve `Connector`'s by `clusterId` so I thought it *might* be useful for `Processor`'s too.

I can remove every such method if the generated (app-services) API is out of scope (leaving only the methods needed for the Data Plane/Control Plane sync). Happy to do so just let me know your thoughts.

Reply:
```
Processors are deployed to the same namespace/cluster as the Connector's to which they can integrate.
```
Do you mean "can be" deployed to the same namespace/cluster? Kafka is the intermediary, so there is nothing from stopping a user from creating a `namespace`  on a completely different Connector Cluster and deploying their `Processors` there. As long as the Processor produces/consumes to the correct Kafka topics, Connectors running in completely different Connector clusters can also consume/produce to the same topics and things should "just work"™. 

```
Happy to do so just let me know your thoughts.
```

I'm sure there must be a use-case for the corresponding Connector method, but I'm not sure what it is yet. Perhaps lets check the synchronizer code to see what it's used for.

Reply:
> I'm sure there must be a use-case for the corresponding Connector method, but I'm not sure what it is yet. 

The generated Connectors method is not used in the RHOC codebase.

IDK if there's another repository with a CLI client?

Reply:
It's not used by {{cos-tools}} (the repository for `rhoc cli`) or `cos-fleetshard`... 

I think it's safe to say it might be redundant.

Reply:
I'm nut 100% sure as naming is not the same but for connectors, there is a rhoc subcommand to get a deployment by id: https://github.com/bf2fc6cc711aee1a0c2a/cos-tools/blob/main/rhoc/pkg/cmd/deployments/get/get.go

Reply:
See https://issues.redhat.com/browse/MGDCTRS-2269